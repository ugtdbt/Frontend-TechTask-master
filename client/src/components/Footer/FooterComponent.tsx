import React, { memo } from "react";
import type { FC } from "react";
import Box from "@mui/material/Box";
import Container from "@mui/material/Container";
import Typography from "@mui/material/Typography";

interface FooterProps {
  description: string;
}

const Footer: FC<FooterProps> = memo((props: FooterProps) => {
  const { description } = props;

  return (
    <Box
      component="footer"
      sx={{ bgcolor: "background.paper", py: 6 }}
      data-testid="footer"
    >
      <Container maxWidth="lg">
        <Typography
          variant="body2"
          color="text.secondary"
          align="center"
          data-testid="footer-description"
        >
          {description}
        </Typography>
      </Container>
    </Box>
  );
});

Footer.displayName = "Footer";
export default Footer;
