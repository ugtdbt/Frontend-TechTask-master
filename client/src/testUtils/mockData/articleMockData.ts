export const article = {
  imageUrl:
    "https://cdn1.home24.net/images/media/catalog/product/200x200/png/l/a/lattenrost-super-fix-90-x-200cm-4448116.webp",
  name: "Lattenrost Super Fix",
  variantName: "Breite: 140 cm",
  price: 14990,
};
