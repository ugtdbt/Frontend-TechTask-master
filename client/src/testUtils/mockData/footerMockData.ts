export const footer = {
  description:
    "Alle Preise sind in Euro (â‚¬) inkl. gesetzlicher Umsatzsteuer und Versandkosten.",
};
