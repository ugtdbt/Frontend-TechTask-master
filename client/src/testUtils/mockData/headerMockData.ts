export const sections = [
  { title: "furniture", url: "#" },
  { title: "lamps", url: "#" },
  { title: "accessories", url: "#" },
  { title: "textiles", url: "#" },
  { title: "children", url: "#" },
  { title: "kitchen", url: "#" },
  { title: "bath", url: "#" },
  { title: "Garden", url: "#" },
  { title: "Brands", url: "#" },
  { title: "Shop the look", url: "#" },
  { title: "advisory", url: "#" },
  { title: "stores", url: "#" },
  { title: "Sale", url: "#" },
];

export const header = {
  sections,
  title: "Home24",
};
