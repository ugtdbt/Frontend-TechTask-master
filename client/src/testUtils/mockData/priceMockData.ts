export const price = {
  price: 2500,
};
