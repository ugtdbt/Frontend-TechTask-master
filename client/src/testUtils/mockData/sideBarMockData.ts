export const sideBar = {
  list: [
    { name: "Wohnzimmer", urlPath: "wohnzimmer-moebel/" },
    { name: "Polsterm√∂bel", urlPath: "alle-polstermoebel/" },
  ],
};
