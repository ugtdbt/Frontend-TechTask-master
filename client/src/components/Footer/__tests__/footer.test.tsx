import { cleanup, render, screen } from "@testing-library/react";
import React from "react";
import renderer from "react-test-renderer";
import { footer } from "../../../testUtils/mockData/footerMockData";
import Footer from "../FooterComponent";

describe("Footer component", () => {
  const FooterComponent = <Footer description={footer.description} />;

  afterEach(() => {
    cleanup();
  });

  it("Footer Match Snapshot", () => {
    const tree = renderer.create(FooterComponent).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Should render Footer component", () => {
    render(FooterComponent);
    const footerElem = screen.getByTestId("footer");
    expect(footerElem).toBeInTheDocument();
  });

  it("Check Footer description", () => {
    render(FooterComponent);
    const footerDescription =
      screen.getByTestId("footer-description")?.innerHTML;
    expect(footerDescription).toContain(footer.description);
  });
});
