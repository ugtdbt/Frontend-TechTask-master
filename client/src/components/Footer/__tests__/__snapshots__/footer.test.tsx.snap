// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component Footer Match Snapshot 1`] = `
<footer
  className="MuiBox-root css-1k76al1"
  data-testid="footer"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter css-1vl95ex-MuiTypography-root"
      data-testid="footer-description"
    >
      Alle Preise sind in Euro (â‚¬) inkl. gesetzlicher Umsatzsteuer und Versandkosten.
    </p>
  </div>
</footer>
`;
